sdk/tests/deqp/modules/shared/glsDrawTests.js

```diff
@@ -1443,59 +1443,7 @@ goog.scope(function() {
     };
 
     /**
-     * @constructor
-     * @param {number} componentCount
-     * @param {deRandom.Random} rnd
-     * @param {glsDrawTests.GLValue} min
-     * @param {glsDrawTests.GLValue} max
-     * @param {boolean} packed
-     */
-    glsDrawTests.RandomArrayGenerator.VertexParameters = function(componentCount, rnd, min, max, packed) {
-        /** @type {number} */ this.componentCount = componentCount;
-        /** @type {deRandom.Random} */ this.rnd = rnd;
-        /** @type {glsDrawTests.GLValue} */ this.min = min;
-        /** @type {glsDrawTests.GLValue} */ this.max = max;
-        /** @type {boolean}*/ this.packed = packed;
-    };
-
-    /**
-     * @param {Array<glsDrawTests.GLValue>} previousComponents
-     * @param {glsDrawTests.RandomArrayGenerator.VertexParameters} vertexParameters
-     * @return {Array<glsDrawTests.GLValue>|Array<number>}
-     */
-    glsDrawTests.RandomArrayGenerator.generateVertex = function(previousComponents, vertexParameters) {
-        /** @type {Array<glsDrawTests.GLValue>|Array<number>} */ var components = [];
-        /** @type {number} */ var limit10 = (1 << 10);
-        /** @type {number} */ var limit2 = (1 << 2);
-
-        for (var componentNdx = 0; componentNdx < vertexParameters.componentCount; componentNdx++) {
-            //keep z and w values the same as the previous attribute
-            if (componentNdx > 1 && previousComponents.length > 2) {
-                components[componentNdx] = previousComponents[componentNdx];
-                continue;
-            }
-
-            components[componentNdx] = vertexParameters.packed ? (
-                componentNdx == 3 ?
-                    new Uint32Array([vertexParameters.rnd.getInt() % limit2])[0] :
-                    new Uint32Array([vertexParameters.rnd.getInt() % limit10])[0]
-            ) : glsDrawTests.GLValue.getRandom(vertexParameters.rnd, vertexParameters.min, vertexParameters.max);
-
-            if (!vertexParameters.packed) {
-                var minSeparation = glsDrawTests.GLValue.minValue(vertexParameters.min.getType());
-
-                // Try to not create vertex near previous
-                if (previousComponents.length > 0 && glsDrawTests.GLValue.abs(components[componentNdx].sub(previousComponents[componentNdx])).lessThan(minSeparation)) {
-                    // Too close, try again (but only once)
-                    components[componentNdx] = glsDrawTests.GLValue.getRandom(vertexParameters.rnd, vertexParameters.min, vertexParameters.max);
-                }
-            }
-        }
-        return components;
-    };
-
-    /**
-     * createBasicArray - Modified JS version to ensure quad generation for compatibility with refrast.
+     * createBasicArray
```

Comment:
Change in this file make the code massively closer to the C++ version, some small differences are noted below